src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLightningRod.java

```diff
@@ -0,0 +1,197 @@
+            }
+
+            if (getOffsetTimer() % 256 == 0 && (aWorld.isThundering() || (aWorld.isRaining() && aXSTR.nextInt(10) == 0))) {
+                int aRodValue = 0;
+                boolean isRodValid = true;
+                int aX = this.getPos().getX();
+                int aY = this.getPos().getY();
+                int aZ = this.getPos().getZ();
+
+                for (int i = this.getPos().getY() + 1; i < aWorld.getHeight()-1; i++) {
+                    if (isRodValid && this.getWorld().getBlockState(new BlockPos(aX, i, aZ)).getBlock().equals(Blocks.IRON_BARS)) {
+                        aRodValue++;
+                    } else {
+                        isRodValid = false;
+                        if (this.getWorld().getBlockState(new BlockPos(aX, i, aZ)).getBlock() != Blocks.AIR) {
+                            aRodValue=0;
+                            break;
+                        }
+                    }
+                }
+                if (!aWorld.isThundering() && ((aY + aRodValue) < 128)) aRodValue = 0;
+                if (aXSTR.nextInt(4 * aWorld.getHeight()) < (aRodValue * (aY + aRodValue))) {
+                    this.energyContainer.addEnergy(getEnergyCapacity() - this.energyContainer.getEnergyStored());
+                    aWorld.addWeatherEffect(new EntityLightningBolt(aWorld, aX, aY + aRodValue, aZ, false));
+                }
+            }
+        }
+    }
+
+    private void setActive(boolean isActive) {
+        if (this.isActive != isActive) {
+            this.isActive = isActive;
+            if (!getWorld().isRemote) {
+                writeCustomData(IS_WORKING, w -> w.writeBoolean(isActive));
+            }
+        }
+    }
+
+    @Override
+    public boolean isActive() {
+        return this.isActive;
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        super.receiveCustomData(dataId, buf);
+        if (dataId == IS_WORKING) {
+            this.isActive = buf.readBoolean();
+        }
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer buf) {
+        super.writeInitialSyncData(buf);
+        buf.writeBoolean(isActive);
+    }
+
+    @Override
+    public void receiveInitialSyncData(PacketBuffer buf) {
+        super.receiveInitialSyncData(buf);
+        this.isActive = buf.readBoolean();
+    }
+
+    @Override
+    protected boolean isEnergyEmitter() {
+        return true;
+    }
+
+    @Override
+    public boolean getIsWeatherOrTerrainResistant(){
+        return true;
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(IGregTechTileEntity iGregTechTileEntity) {
+        return new MetaTileEntityLightningRod(metaTileEntityId, getTier());
+    }
+
+    @Override
+    protected boolean openGUIOnRightClick() {
+        return false;
+    }
+
+    public long getEnergyCapacity() {
+        return 50000000;
+    }
+
+    protected long getMaxInputOutputAmperage() {
+        return 512;
```

Comment:
That seems like a lot.

Reply:
Yes but the recipes is expensive. the HV tier need ZPM voltage materials. so she is not imba.